src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2197,3 +2227,14 @@ type GetResponseFound<TDocument = unknown> = estypes.GetResponse<TDocument> &
 const isFoundGetResponse = <TDocument = unknown>(
   doc: estypes.GetResponse<TDocument>
 ): doc is GetResponseFound<TDocument> => doc.found;
+
+/**
+ * Response headers check to determine if the response is from Elasticsearch
+ * @param headers Response headers
+ * @returns boolean
+ */
+const isEsHeaderValid = (headers: Record<string, unknown> | null) => {
```

Comment:
question: What if `isSupportedEsServer`? `header` is an implementation detail.

Reply:
I don't understand what you're asking with "What if isSupportedEsServer?". Would you mind explaining a bit please?

Reply:
sorry, it should be `What if we name it "isSupportedEsServer"`